### DISCUSSION TITLE
SMLets.Exchange.Connector.Resources.Runscript error

### DISCUSSION THREAD

**Discussion by JustinRCron** [AdhocAdam/smletsexchangeconnector#471] [Q&A] [answered]
![image](https://github.com/AdhocAdam/smletsexchangeconnector/assets/141958301/8d7ce0c6-a2a7-4308-b0c6-6c5b32de3b5d)
I am getting this error and I could use some assistance in fixing it. I am not 100% sure what I am supposed to be adding to the Set-PSRepository cmdlet, it asks for a name.

Anybody else experience this issue and know how to fix it?

Any help would be greatly appreciated.

Thank you!

**Comment by AdhocAdam** [comment, marked as answer]
The only place I can think that this would be happening is if SMLets is not detected on the server, and then attempts to install itself from the PowerShell Gallery. In which case, open an elevated PowerShell, install it, and then let the connector run again.

```powershell
Install-Module SMLets
```

If this works and the error resolves, then I can look into introducing more logic around the installation to prevent this from happening for others.

**Comment by JustinRCron** [comment]
Thank you Adam, that fixed it and it is now processing emails.

I greatly appreciate your help!

**Reply by AdhocAdam** [reply]
Most excellent to hear! If the connector has helped you out and do more cool stuff, consider Starring the repo 😃
